apps/dav/lib/CalDAV/Reminder/Backend.php

```diff
@@ -64,11 +64,12 @@ public function __construct(IDBConnection $db,
 	 */
 	public function getRemindersToProcess():array {
 		$query = $this->db->getQueryBuilder();
-		$query->select(['cr.*', 'co.calendardata', 'c.displayname', 'c.principaluri'])
+		$query->select(['cr.*', 'co.calendardata', 'c.displayname', 'c.principaluri','cr.notification_date', 'cr.event_hash'])
 			->from('calendar_reminders', 'cr')
 			->where($query->expr()->lte('cr.notification_date', $query->createNamedParameter($this->timeFactory->getTime())))
 			->join('cr', 'calendarobjects', 'co', $query->expr()->eq('cr.object_id', 'co.id'))
-			->join('cr', 'calendars', 'c', $query->expr()->eq('cr.calendar_id', 'c.id'));
+			->join('cr', 'calendars', 'c', $query->expr()->eq('cr.calendar_id', 'c.id'))
+			->groupBy('cr.notification_date', 'cr.event_hash');
```

Comment:
```suggestion
			->groupBy('cr.event_hash', 'cr.notification_date');
```

Yep. If the hash and the date is equal we fully deduplicate.
If the hash differs but the date is equal we leave the duplicates.